sql/decimal.go

```diff
@@ -212,6 +215,34 @@ func (t decimalType) Zero() interface{} {
 	return decimal.NewFromInt(0).StringFixed(int32(t.scale))
 }
 
+// Marshal takes a valid Decimal value and returns it as an string.
+func (t decimalType) Marshal(v interface{}) (string, error) {
```

Comment:
So as I mentioned on the dolt PR, I don't like this representation. You're pushing storage concerns like sort order onto a layer of the code where it doesn't belong. This marshall and unmarshall code belongs in the Dolt (or other storage layer) codebase, not here. Parsing values to and from a string for input / output is a legitimate concern, but this goes much farther. Rather, this code should define a storage-agnostic format for passing values, like a simple struct that mirrors the values found in the shopspring.decimal code. Let the storage layer concern itself with how to store these things.

(This next comment really belongs in the Dolt PR for the reason above, but since the code is here the comment is here). As for how to store them: strings are not a good fit. The zero padding wastes space, and the representation of negative numbers requires you to know the scale to unmarshall a value. Ideally these values can all be evaluated independent of the type params. You probably want to implement a new type in Noms for this, basically copying big.int and shopspring.decimal, and then implement a .Less function for the new type. Your implementation shouldn't be much slower than comparing strings if you do it right.